src/spikeinterface/curation/auto_merge.py

```diff
@@ -369,19 +368,126 @@ def compute_merge_unit_groups(
+        for key, pair in new_merges.items():
+            nested_merge = np.flatnonzero([i in pair for i in old_keys])
+            if len(nested_merge) == 0:
+                resolved_merges.update({key: pair})
+            else:
+                for n in nested_merge:
+                    previous_merges = resolved_merges.pop(old_keys[n])
+                    pair.remove(old_keys[n])
+                    pair += previous_merges
+                    resolved_merges.update({key: pair})
+        return resolved_merges
+
+
+def _auto_merge_units_single_iteration(
+    sorting_analyzer: SortingAnalyzer,
+    compute_merge_kwargs: dict = {},
+    apply_merge_kwargs: dict = {},
+    extra_outputs: bool = False,
+    force_copy: bool = True,
+    merging_strategy: str = "warning",
+    **job_kwargs,
 ) -> SortingAnalyzer:
     """
-    Compute merge unit groups and apply it on a SortingAnalyzer.
+    Compute merge unit groups and apply it on a SortingAnalyzer. Used by auto_merge_units, see documentation there.
     Internally uses `compute_merge_unit_groups()`
+
+    Parameters
+    ----------
+
+    sorting_analyzer : SortingAnalyzer
+        The SortingAnalyzer to be merged
+    compute_merge_kwargs : dict
+        The parameters to be passed to compute_merge_unit_groups()
+    apply_merge_kwargs : dict
+        The parameters to be passed to merge_units()
+    extra_outputs : bool, default: False
+        If True, additional outputs are returned
+    force_copy : bool, default: True
+        If True, the sorting_analyzer is copied before applying the merges
+    merging_strategy : str, default: "warning"
+        The strategy to use when merging units. Can be "warning" or "strict". If "warning", the merging will
+        stop with a warning if soft merges can not be done. Otherwise, if "strict", an error is thrown
+
+    Returns
+    -------
+    sorting_analyzer:
+        The new sorting analyzer where all the merges from all the presets have been applied
+
+    resolved_merges, merge_unit_groups, outs:
+        Returned only when extra_outputs=True
+        A list with the merges performed, and dictionaries that contains data for debugging and plotting.
     """
+
+    if force_copy:
+        # To avoid erasing the extensions of the user
+        sorting_analyzer = sorting_analyzer.copy()
+
     merge_unit_groups = compute_merge_unit_groups(
-        sorting_analyzer, extra_outputs=False, **compute_merge_kwargs, **job_kwargs
+        sorting_analyzer, **compute_merge_kwargs, extra_outputs=extra_outputs, force_copy=False, **job_kwargs
     )
 
-    merged_analyzer = sorting_analyzer.merge_units(merge_unit_groups, **apply_merge_kwargs, **job_kwargs)
-    return merged_analyzer
+    if extra_outputs:
+        merge_unit_groups, outs = merge_unit_groups
+
+    merged_units = len(merge_unit_groups) > 0
+    if merged_units:
+        is_mergeable = sorting_analyzer.is_mergeable(merge_unit_groups, **apply_merge_kwargs)
```

Comment:
See comment above. If the `sorting_analyzer.is_mergeable` returned a dictionary with mergeability for each merge_unit_group, then here we can be much more "surgical" and just skip groups that cannot be merged. What do you think @yger ?

Reply:
This has been implemented. Now the functions are_units_mergeables return a dict whose keys are the new unit ids, and values boolean telling if the merge is possible or not